spatialmath/box.go

```diff
@@ -352,3 +355,95 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector. This method takes one argument which
+// determines how many points to place per square mm. If the argument is set to 0. we
+// automatically substitute the value with defaultBoxPointDensity.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+
+	// TODO: the fillFaces method calls can be made concurrent if the ToPointCloud method is too slow
+	// the boolean values which are passed into the fillFaces method allow for common edges to be
+	// ignored. This removes duplicate points on the box's surface
+	var facePoints [][]r3.Vector
+	facePoints = append(facePoints, fillFaces(b.halfSize, iter, 0, true, false))
+	facePoints = append(facePoints, fillFaces(b.halfSize, iter, 1, true, true))
+	facePoints = append(facePoints, fillFaces(b.halfSize, iter, 2, false, false))
+
+	transformedVecs := transformPointsToPose(facePoints, b.Pose())
+	return transformedVecs
+}
+
+// fillFaces returns a list of vectors which lie on the surface of the box.
+func fillFaces(halfSize [3]float64, iter float64, fixedDimension int, orEquals1, orEquals2 bool) []r3.Vector {
+	var facePoints []r3.Vector
+	// create points on box faces with box centered at (0, 0, 0)
+	starts := [3]float64{0.0, 0.0, 0.0}
+	// depending on which faces we want to fill, we must keep either i xor j xor k constant
```

Comment:
want to double check that the comment here should continue to say "xor" and not be "or"

Reply:
xor is correct, but it might be more readable to say "depending on which face we want to fill, one of i,j,k is kept constant"